web-access-app/src/main/java/edu/unc/lib/boxc/web/access/controllers/FullRecordController.java

```diff
@@ -260,6 +262,110 @@ public String getFullRecord(String pidString, Model model, HttpServletRequest re
         return "fullRecord";
     }
 
+    /**
+     * JSON representation of full record
+     * Can remove the non-JSON version of the full record once all JSP files are ported over
+     * @param pidString
+     * @return
+     */
+    @GetMapping(path = "/{pid}/json", produces = APPLICATION_JSON_VALUE)
+    public @ResponseBody String handleRequest(@PathVariable("pid")String pidString) {
+        PID pid = PIDs.get(pidString);
+
+        AccessGroupSet principals = getAgentPrincipals().getPrincipals();
+        aclService.assertHasAccess("Insufficient permissions to access full record for " + pidString,
+                pid, principals, Permission.viewMetadata);
+
+        // Retrieve the object's record from Solr
+        SimpleIdRequest idRequest = new SimpleIdRequest(pid, principals);
+        ContentObjectRecord briefObject = queryLayer.getObjectById(idRequest);
+
+        if (briefObject == null) {
+            throw new NotFoundException("No record found for " + pid.getId());
+        }
+
+        // Get additional information depending on the type of object since the user has access
+        String resourceType = briefObject.getResourceType();
+        if (!ResourceType.File.nameEquals(resourceType)) {
+            briefObject.getCountMap().put("child", childrenCountService.getChildrenCount(briefObject, principals));
+        }
+
+        HashMap<String, Object> recordProperties = new HashMap<>();
+        recordProperties.put("briefObject", briefObject);
+        recordProperties.put("resourceType", resourceType);
+
+        // Get parent id
+        if (ResourceType.File.nameEquals(resourceType)) {
```

Comment:
We could resolve the cognitive complexity issues in this method pretty easily (and make it a lot easier to read) by breaking some of these long conditionals wrapping property insertions into their own private helper methods. Like `addContainingWorkUUID` or `addContainingWorkUUIDForFile` for this one.